copy the correct version of LLVM into the stage0 sysroot

onur-ozkan opened the pull request:
In some cases(see https://github.com/rust-lang/rust/issues/109314), when the stage0
compiler relies on more recent version of LLVM than the beta compiler, it may not
be able to locate the correct LLVM in the sysroot. This situation typically occurs
when we upgrade LLVM version while the beta compiler continues to use an older version.

Fixes #109314

Mark-Simulacrum commented on src/bootstrap/compile.rs:
```diff
@@ -1390,6 +1390,18 @@ impl Step for Sysroot {
         let _ = fs::remove_dir_all(&sysroot);
         t!(fs::create_dir_all(&sysroot));
 
+        // When bootstrap relies on a more recent version of LLVM than the beta compiler,
```
"bootstrap relies on a more recent version of LLVM" -- can you say more about what you mean by relies on here? How are we relying on LLVM from bootstrap? The linked issue seems like it's not about bootstrap relying but rather that stage0 compiler tests do, if I intuited right at a glance?

(Code seems OK, but a comment update would be good).

onur-ozkan committed:
add the correct version of LLVM into the stage0 sysroot

In some cases(see https://github.com/rust-lang/rust/issues/109314), when the stage0
compiler relies on more recent version of LLVM than the beta compiler, it may not
be able to locate the correct LLVM in the sysroot. This situation typically occurs
when we upgrade LLVM version while the beta compiler continues to use an older version.

Signed-off-by: ozkanonur <work@onurozkan.dev>

Mark-Simulacrum commented:
@bors r+ rollup=iffy

I'm going to go ahead and approve but let's keep an eye out for any regressions and back out if needed.

bors commented:
:pushpin: Commit aec4b59385e6a2d5c4bee675691b369b30ad5700 has been approved by `Mark-Simulacrum`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=Mark-Simulacrum aec4b59385e6a2d5c4bee675691b369b30ad5700 -->
<!-- homu: {"type":"Approved","sha":"aec4b59385e6a2d5c4bee675691b369b30ad5700","approver":"Mark-Simulacrum","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

bors commented:
:hourglass: Testing commit aec4b59385e6a2d5c4bee675691b369b30ad5700 with merge ebbd7154a713a282e4b39da5107cdae2bf2efcb4...
<!-- homu: {"type":"BuildStarted","head_sha":"aec4b59385e6a2d5c4bee675691b369b30ad5700","merge_sha":"ebbd7154a713a282e4b39da5107cdae2bf2efcb4"} -->

bors commented:
:sunny: Test successful - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/5848955862/job/15857630379)
Approved by: Mark-Simulacrum
Pushing ebbd7154a713a282e4b39da5107cdae2bf2efcb4 to master...
<!-- homu: {"type":"BuildCompleted","approved_by":"Mark-Simulacrum","base_ref":"master","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/actions/runs/5848955862/job/15857630379"},"merge_sha":"ebbd7154a713a282e4b39da5107cdae2bf2efcb4"} -->

bors commented:
:sunny: Test successful - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/5848955862/job/15857630379)
Approved by: Mark-Simulacrum
Pushing ebbd7154a713a282e4b39da5107cdae2bf2efcb4 to master...
<!-- homu: {"type":"BuildCompleted","approved_by":"Mark-Simulacrum","base_ref":"master","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/actions/runs/5848955862/job/15857630379"},"merge_sha":"ebbd7154a713a282e4b39da5107cdae2bf2efcb4"} -->

rust-timer commented:
Finished benchmarking commit (ebbd7154a713a282e4b39da5107cdae2bf2efcb4): [comparison URL](https://perf.rust-lang.org/compare.html?start=644e8068cbffef6428d1970fe50a4a863079ee54&end=ebbd7154a713a282e4b39da5107cdae2bf2efcb4&stat=instructions:u).

### Overall result: ❌ regressions - no action needed


@rustbot label: -perf-regression



### [Instruction count](https://perf.rust-lang.org/compare.html?start=644e8068cbffef6428d1970fe50a4a863079ee54&end=ebbd7154a713a282e4b39da5107cdae2bf2efcb4&stat=instructions:u)
This is a highly reliable metric that was used to determine the overall result at the top of this comment.

|                                    | mean | range        | count |
|:----------------------------------:|:----:|:------------:|:-----:|
| Regressions ❌ <br /> (primary)    | -    | -            | 0     |
| Regressions ❌ <br /> (secondary)  | 0.4% | [0.4%, 0.4%] | 2     |
| Improvements ✅ <br /> (primary)   | -    | -            | 0     |
| Improvements ✅ <br /> (secondary) | -    | -            | 0     |
| All ❌✅ (primary)                 | -    | -            | 0     |

### [Max RSS (memory usage)](https://perf.rust-lang.org/compare.html?start=644e8068cbffef6428d1970fe50a4a863079ee54&end=ebbd7154a713a282e4b39da5107cdae2bf2efcb4&stat=max-rss)
<details>
<summary>Results</summary>

This is a less reliable metric that may be of interest but was not used to determine the overall result at the top of this comment.

|                                    | mean  | range          | count |
|:----------------------------------:|:-----:|:--------------:|:-----:|
| Regressions ❌ <br /> (primary)    | 4.1%  | [2.9%, 5.3%]   | 2     |
| Regressions ❌ <br /> (secondary)  | 2.1%  | [1.8%, 2.4%]   | 2     |
| Improvements ✅ <br /> (primary)   | -2.6% | [-2.6%, -2.6%] | 1     |
| Improvements ✅ <br /> (secondary) | -3.3% | [-3.4%, -3.2%] | 2     |
| All ❌✅ (primary)                 | 1.9%  | [-2.6%, 5.3%]  | 3     |
</details>

### [Cycles](https://perf.rust-lang.org/compare.html?start=644e8068cbffef6428d1970fe50a4a863079ee54&end=ebbd7154a713a282e4b39da5107cdae2bf2efcb4&stat=cycles:u)
This benchmark run did not return any relevant results for this metric.

### [Binary size](https://perf.rust-lang.org/compare.html?start=644e8068cbffef6428d1970fe50a4a863079ee54&end=ebbd7154a713a282e4b39da5107cdae2bf2efcb4&stat=size:linked_artifact)
This benchmark run did not return any relevant results for this metric.

**Bootstrap**: 632.481s -> 631.811s (-0.11%)
